test/geometry/LinearPropagator.test.cc

```diff
@@ -84,19 +84,23 @@ TEST_F(LinearPropagatorHostTest, track_line)
 
         geo.find_next_step();
```

Comment:
With this interface change, one of the goals is to get rid of the requirement to interact directly with the geometry when performing propagation. I think this call should be moved into the propagator itself, since it will be different between the linear propagator and the magnetic field propagator.